### PR TITLE
Fix maven, Fix ctz_npc_id function, Add ctz_npc_click event

### PR DESCRIPTION
**Pull request by entrypointkr**
In my case, I couldn't get the npc id(number) by uuid of bukkit entity using ctz_npc_id().
So I tried to know why it happens and I known that citizen's npc entities are not be register to bukkit entities map. so we can't get an npc entity using `Bukkit.getEntity(uuid)`
And not equals bukkit entity's uuid and citizen's internal uuid. so weired.
So I added ctz_npc_click event and fix ctz_npc_id to do not gettering entity.
It works fine in CH 3.3.4, Citizens-2.0.25-b1738 latest versions

### CODE REVIEW

File: src/main/java/com/hekta/chcitizens/core/events/CitizensEvents.java [hunk 1]
```diff
@@ -293,4 +297,59 @@ public boolean matches(Map<String, Mixed> prefilter, BindableEvent event) throws
...
+			if (e instanceof MCCitizensNPCClickEvent) {
+				Target t = Target.UNKNOWN;
+				MCCitizensNPCClickEvent event = ((MCCitizensNPCClickEvent) e);
+				Map<String, Mixed> ret = new HashMap<>();
+				boolean left = event.isLeft();
+				MCCitizensNPC npc = event.getNPC();
+				MCEntity entity = npc.getEntity();
+				ret.put("button", new CString(left ? "left" : "right", t));
+				ret.put("player", new CString(event.getClicker().getName(), t));
+				ret.put("id", new CInt(npc.getId(), t));
```
Comment (PseudoKnight) [inline src/main/java/com/hekta/chcitizens/core/events/CitizensEvents.java]: This should probably be "npc" to match the other events.

**Comment by PseudoKnight** [comment]
Regarding entities map, NPCs used to be added to that, but I guess that changed. This new implementation is better anyway. It's faster and cleaner.

**Comment by entrypointkr** [comment]
Yea but I don't know why unable to get a NPC by entity uuid using `CitizensAPI.getNPCRegistry().getByUniqueId()` or `CitizensAPI.getNPCRegistry().getByUniqueIdGlobal()`.
Not equals between NPC's internal uuid and bukkit entity uuid. likes below.
If bukkit entity `1234-1234-1234-1234`,
citizen's npc one `1234-1234-3234-1234`
Just one character is difference.
Is this for prevent uuid duplicate? I don't know
Also we can't use `ctz_entity_is_npc()`

But it doesn't matter. we can use NPCClickEvent to get npc's id for npc identify :) Thanks for review

**Comment by PseudoKnight** [comment]
Hm. While I've taken over maintenance of this extension, I no longer actively use it on my server. It sounds like I may need to test and improve some things.

Thanks for this.

**Pull request merged**